### PR TITLE
Apply "cppcoreguidelines-pro-type-member-init" clang-tidy check

### PR DESCRIPTION
**Pull request by florian-kuebler**
Test: Compile

### CODE REVIEW

File: src/LinuxCaptureService/LinuxCaptureServiceBase.cpp [hunk 1]
```diff
@@ -183,7 +183,7 @@ struct TargetProcessStateAfterCapture {
 
 TargetProcessStateAfterCapture GetTargetProcessStateAfterCapture(
     pid_t pid, const absl::flat_hash_set<std::string>& old_core_files) {
-  TargetProcessStateAfterCapture result;
+  TargetProcessStateAfterCapture result{};
   result.process_state = CaptureFinished::kProcessStateInternalError;
   result.termination_signal = CaptureFinished::kTerminationSignalInternalError;
```
Comment (dpallotti) [inline src/LinuxCaptureService/LinuxCaptureServiceBase.cpp:188]: Use designated initializers.

File: src/OrbitBase/include/OrbitBase/Profiling.h [hunk 2]
```diff
@@ -51,7 +51,7 @@ namespace orbit_base {
 constexpr clockid_t kOrbitCaptureClock = CLOCK_MONOTONIC;
 
 [[nodiscard]] inline uint64_t CaptureTimestampNs() {
-  timespec ts;
+  timespec ts{};
```
Comment (dpallotti) [inline src/OrbitBase/include/OrbitBase/Profiling.h:54]: This also seems uncalled for as this is an output parameter of `clock_gettime` that follows.

File: src/UserSpaceInstrumentation/TrampolineTest.cpp [hunk 3]
```diff
@@ -679,15 +679,15 @@ class InstrumentFunctionTest : public testing::Test {
...
+  uint64_t trampoline_address_{};
+  uint64_t return_trampoline_address_{};
+  uint64_t entry_payload_function_address_{};
+  uint64_t exit_payload_function_address_{};
 
   absl::flat_hash_map<uint64_t, uint64_t> relocation_map_;
 
   std::string function_name_;
-  uint64_t function_address_;
+  uint64_t function_address_{};
```
Comment (dpallotti) [inline src/UserSpaceInstrumentation/TrampolineTest.cpp:690]: The fields above use `= 0`, so we should be consistent.

File: src/UserSpaceInstrumentation/RegisterState.cpp [hunk 4]
```diff
@@ -123,7 +123,7 @@ ErrorMessageOr<void> RegisterState::RestoreRegisters() {
   // BackupRegisters needs to be called before RestoreRegisters.
   ORBIT_CHECK(tid_ != -1);
 
-  iovec iov;
+  iovec iov{};
   iov.iov_base = &general_purpose_registers_;
   iov.iov_len = (bitness_ == Bitness::k32Bit) ? sizeof(GeneralPurposeRegisters32)
                                               : sizeof(GeneralPurposeRegisters64);
```
Comment (dpallotti) [inline src/UserSpaceInstrumentation/RegisterState.cpp:129]: Use designated initializers.

File: src/UserSpaceInstrumentation/RegisterState.cpp [hunk 5]
```diff
@@ -80,7 +80,7 @@ bool RegisterState::HasAvxDataStored() {
 ErrorMessageOr<void> RegisterState::BackupRegisters(pid_t tid) {
   tid_ = tid;
 
-  iovec iov;
+  iovec iov{};
   iov.iov_base = &general_purpose_registers_;
   iov.iov_len = sizeof(GeneralPurposeRegisters);
```
Comment (dpallotti) [inline src/UserSpaceInstrumentation/RegisterState.cpp:85]: Use designated initializers.

File: src/LinuxTracing/PerfEvent.h [hunk 6]
```diff
@@ -35,7 +35,7 @@ perf_event_sample_regs_user_all_to_register_array(const perf_event_sample_regs_u
 // while the `data` field holds the data specific to the individual event.
 template <typename PerfEventDataT>
 struct TypedPerfEvent {
-  uint64_t timestamp;
+  uint64_t timestamp{};
```
Comment (dpallotti) [inline src/LinuxTracing/PerfEvent.h:38]: Personal preference across the change: For member variables I prefer explicit values (e.g., `0`, `nullptr`, `false`). Also the next field is initialized more explicitly.

**Review by dpallotti — COMMENTED** [review]
I think this change is too mechanical: some of the non-initializations were on purpose, while others we would probably write differently.

File: src/ClientData/ScopeTreeTimerDataTest.cpp [hunk 7]
```diff
@@ -36,7 +36,7 @@ constexpr uint64_t kMinTimestamp = 2;
 constexpr uint64_t kMaxTimestamp = 11;
 
 TimersInTest AddTimersInScopeTreeTimerDataTest(ScopeTreeTimerData& scope_tree_timer_data) {
-  TimersInTest inserted_timers;
+  TimersInTest inserted_timers{};
```
[the remark this answers is not in the document]
Comment (florian-kuebler) [reply src/ClientData/ScopeTreeTimerDataTest.cpp]: changed it.

File: src/LinuxCaptureService/LinuxCaptureServiceBase.cpp (hunk 1 above)
Comment (florian-kuebler) [reply src/LinuxCaptureService/LinuxCaptureServiceBase.cpp]: Done.

File: src/ClientData/include/ClientData/ThreadStateSliceInfo.h [hunk 8]
```diff
@@ -69,7 +69,7 @@ class ThreadStateSliceInfo {
   WakeupReason wakeup_reason_;
   uint32_t wakeup_tid_;
   uint32_t wakeup_pid_;
-  std::optional<uint64_t> switch_out_or_wakeup_callstack_id_;
+  std::optional<uint64_t> switch_out_or_wakeup_callstack_id_{};
```
[the remark this answers is not in the document]
Comment (florian-kuebler) [reply src/ClientData/include/ClientData/ThreadStateSliceInfo.h]: Yeah that is strange, changed it back.

File: src/LinuxTracing/PerfEvent.h (hunk 6 above)
Comment (florian-kuebler) [reply src/LinuxTracing/PerfEvent.h]: Agree. Changed it here and hopefully across the pr.

File: src/LinuxTracing/PerfEventReaders.cpp [hunk 9]
```diff
@@ -256,10 +256,10 @@ MmapPerfEvent ConsumeMmapPerfEvent(PerfEventRingBuffer* ring_buffer,
   // };
   // Because of filename, the layout is not fixed.
 
-  perf_event_sample_id_tid_time_streamid_cpu sample_id;
+  perf_event_sample_id_tid_time_streamid_cpu sample_id{};
```
[the remark this answers is not in the document]
Comment (florian-kuebler) [reply src/LinuxTracing/PerfEventReaders.cpp]: Agree. Removed it here and elsewhere in the file.

File: src/OrbitBase/include/OrbitBase/Profiling.h (hunk 2 above)
Comment (florian-kuebler) [reply src/OrbitBase/include/OrbitBase/Profiling.h]: Done.

File: src/UserSpaceInstrumentation/RegisterState.cpp (hunk 5 above)
Comment (florian-kuebler) [reply src/UserSpaceInstrumentation/RegisterState.cpp]: Done.

File: src/UserSpaceInstrumentation/RegisterState.cpp (hunk 4 above)
Comment (florian-kuebler) [reply src/UserSpaceInstrumentation/RegisterState.cpp]: Done.

File: src/UserSpaceInstrumentation/TrampolineTest.cpp (hunk 3 above)
Comment (florian-kuebler) [reply src/UserSpaceInstrumentation/TrampolineTest.cpp]: Done.

File: src/UserSpaceInstrumentation/TrampolineTest.cpp [hunk 10]
```diff
@@ -679,15 +679,15 @@ class InstrumentFunctionTest : public testing::Test {
...
+  uint64_t trampoline_address_ = 0;
+  uint64_t return_trampoline_address_ = 0;
+  uint64_t entry_payload_function_address_ = 0;
+  uint64_t exit_payload_function_address_ = 0;
 
   absl::flat_hash_map<uint64_t, uint64_t> relocation_map_;
 
   std::string function_name_;
-  uint64_t function_address_;
+  uint64_t function_address_{};
```
Comment (dpallotti) [inline src/UserSpaceInstrumentation/TrampolineTest.cpp:690]:
```suggestion
  uint64_t function_address_ = 0;
```
As per previous comment on consistency with the other fields.

**Review by dpallotti — APPROVED** [review]

File: src/UserSpaceInstrumentation/TrampolineTest.cpp (hunk 10 above)
Comment (florian-kuebler) [reply src/UserSpaceInstrumentation/TrampolineTest.cpp:690]: Done.

**Pull request merged**